v language shared array slice error!

### Describe the bug

v language uses a shared array, the shared array slice operation, running code error.
```v
// shared_array.v:
fn main() {
    shared a := ["a", "b"]

    a << ["c", "d"]
    println("a: ${ rlock{ a } }") //print: ["a","b","c","d"]

    a3 := rlock{ a[3] } //print: d
    println("a3: ${ rlock{ a3 } }")

    // The following two lines run an error:
    a_slice := rlock{ a[1..3] } //Slice gone wrong
    println("a[1..3]: ${ rlock{ a_slice } }") //expect: ["b", "c"]
}
```
### Reproduction Steps
```v
fn main() {
    shared a := ["a", "b"]

    a << ["c", "d"]
    println("a: ${ rlock{ a } }") //print: ["a","b","c","d"]

    a3 := rlock{ a[3] } //print: d
    println("a3: ${ rlock{ a3 } }")

    // The following two lines run an error:
    a_slice := rlock{ a[1..3] } //Slice gone wrong
    println("a[1..3]: ${ rlock{ a_slice } }") //expect: ["b", "c"]
}
```
### Expected Behavior

Normal should be able to take the slice and output the correct result. For example, this example should output correctly:
a[1..3]: ["b", "c"]

### Current Behavior

The run error message is as follows:
```
~ $ v run shared_array.v
================== C compilation error (from cc): ==============
cc: /data/data/com.termux/files/usr/tmp/v_10264/shared_array.01KJEX8FMH7VYB2SPMPKQHDZD0.tmp.c:8820:7: error: assigning to '__shared__Array_string' (aka 'struct __shared__Array_string') from incompatible type 'array' (aka 'struct array')
cc:  8820 |                 _t5 = builtin__array_slice((*a).val, 1, 3);
cc:       |                     ^ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
cc: /data/data/com.termux/files/usr/tmp/v_10264/shared_array.01KJEX8FMH7VYB2SPMPKQHDZD0.tmp.c:8824:29: error: member reference type '__shared__Array_string' (aka 'struct __shared__Array_string') is not a pointer; did you mean to use '.'?
cc:  8824 |                 Array_string a_slice = _t5->val;
cc:       |                                        ~~~^~
cc:       |                                           .
cc: 2 errors generated.
================================================================
Try passing `-g` when compiling, to see a .v file:line information, that correlates more with the C error.
(Alternatively, pass `-show-c-output`, to print the full C error message).
builder error:
==================
```
### Possible Solution

_No response_

### Additional Information/Context

_No response_

### V version

V 0.5.0 57e617a

### Environment details (OS name and version, etc.)
```
|V full version      |V 0.5.0 a8b0e1d5ed3096edce359cd5d7d2fbb7ab50d31e.57e617a
|:-------------------|:-------------------
|OS                  |termux, 5.10.136-android12-9-00020-gc9f59ef34367-ab9585114, #1 SMP PREEMPT Wed Feb 8 23:50:57 UTC 2023
|Processor           |8 cpus, 64bit, little endian
|Memory              |N/A
|                    |
|V executable        |/data/data/com.termux/files/home/v/v
|V last modified time|2026-02-27 07:02:36
|                    |
|V home dir          |OK, value: /data/data/com.termux/files/home/v
|VMODULES            |OK, value: /data/data/com.termux/files/home/.vmodules
|VTMP                |OK, value: /data/data/com.termux/files/usr/tmp/v_10264
|Current working dir |OK, value: /data/data/com.termux/files/home
|                    |
|Git version         |git version 2.51.0
|V git status        |57e617aa-dirty
|.git/config present |true
|                    |
|cc version          |clang version 20.1.8
|gcc version         |clang version 20.1.8
|clang version       |Android (dev, +pgo, -bolt, +lto, -mlgo, based on r522817b) clang version 18.0.2 (https://android.googlesource.com/toolchain/llvm-project d8003a456d14a3deb8054cdaa529ffbf02d9b262)
|tcc version         |N/A
|tcc git status      |thirdparty-unknown-unknown de82a130
|emcc version        |N/A
|glibc version       |readelf: Error: 'readlink (GNU coreutils) 9.7
```
> [!NOTE]
> You can use the 👍 reaction to increase the issue's priority for developers.
>
> Please note that only the 👍 reaction to the issue itself counts as a vote.
> Other reactions and those to comments will not be taken into account.

--- a/vlib/v/checker/checker.v
+++ b/vlib/v/checker/checker.v
@@ -5063,6 +5063,32 @@ fn (mut c Checker) lock_expr(mut node ast.LockExpr) ast.Type {
 		mut last_stmt := node.stmts.last()
 		if mut last_stmt is ast.ExprStmt {
 			c.expected_type = expected_type
+			// Check for shared array slice - auto clone for safety
+			if mut last_stmt.expr is ast.IndexExpr {
+				index_expr := last_stmt.expr
+				if index_expr.index is ast.RangeExpr && index_expr.left_type.has_flag(.shared_f)
+					&& !c.inside_unsafe {
+					// Slicing a shared array creates a view over shared memory.
+					// Auto-clone for safety to avoid data races.
+					c.add_error_detail_with_pos('To silence this notice, use either an explicit `.clone()`,
+or use an explicit `unsafe{ ... }` block, if you do not want a copy of the slice.',
+						index_expr.pos)
+					c.note('an implicit clone of the shared array slice was done here',
+						index_expr.pos)
+					slice_type := index_expr.typ
+					last_stmt.expr = ast.CallExpr{
+						name:           'clone'
+						kind:           .clone
+						left:           index_expr
+						left_type:      slice_type
+						is_method:      true
+						receiver_type:  slice_type
+						return_type:    slice_type
+						scope:          c.fn_scope
+						is_return_used: true
+					}
+				}
+			}
 			ret_type = c.expr(mut last_stmt.expr)
 		}
 	}

--- a/vlib/v/tests/concurrency/shared_lock_expr_test.v
+++ b/vlib/v/tests/concurrency/shared_lock_expr_test.v
@@ -100,9 +100,11 @@ fn test_shared_lock_array_index_expr() {
 
 fn test_shared_lock_array_slice_expr() {
 	// Test slicing shared arrays in rlock expressions (fixes issue #26663)
+	// Note: slicing a shared array automatically clones for safety,
+	// since a slice is a view over the same memory buffer.
 	shared a := ['a', 'b', 'c', 'd']
 
-	// Basic slice
+	// Basic slice - implicit clone happens (safe independent copy)
 	slice1 := lock {
 		a[1..3]
 	}
@@ -142,6 +144,18 @@ fn test_shared_lock_array_slice_expr() {
 	assert slice5[0] == 2
 	assert slice5[1] == 3
 	assert slice5[2] == 4
+
+	// Explicit clone - same behavior, no warning
+	slice6 := lock {
+		a[1..3].clone()
+	}
+	assert slice6 == ['b', 'c']
+
+	// Unsafe block - get a view (user takes responsibility)
+	slice7 := lock {
+		unsafe { a[1..3] }
+	}
+	assert slice7 == ['b', 'c']
 }
 
 struct DummySt {

